scss/bitstyles/tools/_color.scss

```diff
@@ -0,0 +1,52 @@
+@use "sass:map";
+@use "sass:math";
+
+// Generate palette
+//
+// **Generates a Sass map of colors. Each color is a `name: color` pair, with the name being the percentage as a string (one for each of the percentages in the list you pass in), and the color being the mix-color mixed into the base-color to that percentage.
+//
+// @param {String} $base-color  Base colour into which the other color will be mixed
+//
+// @param {String} $mix-color  The color to mix into the base color at different percentages
+//
+// @param {list} $percentages  A list of decmial values representing the percentages at which the mix-color will be mixed into the base-color
+
+@function generate-palette($base-color, $mix-color, $percentages) {
+  $color-map: ();
+
+  @each $percentage in $percentages {
+    $color-map: map.merge($color-map, ('#{$percentage}': mix($mix-color, $base-color, 100% - percentage(math.div($percentage, 100)))));
```

Comment:
✅  reads good. I wonder if there is such a thing as tests for the logic-parts of sass, but I guess that is not a thing 🤔

Reply:
Our oldest open issue is related to sass unit tests :D (#53) It felt like overkill at the time, but now that we’re generating more CSS using functions, it could be worth investigating again 👍